test/FunctionalTests/Microsoft.OData.Client.Tests/Tracking/DataServiceContextTrackOnlyMleNonAsyncApiTests.cs

```diff
@@ -0,0 +1,335 @@
+      ""Id"": 1,
+      ""Name"": ""Sample Doc 1"",
+      ""FileLength"": 0
+    },
+   {
+      ""@odata.mediaReadLink"": ""https://localhost:8000/Documents/2/content"",
+      ""@odata.mediaContentType"": ""application/pdf"",
+      ""Id"": 2,
+      ""Name"": ""Sample Doc 2"",
+      ""FileLength"": 0
+    },
+    {
+      ""@odata.mediaReadLink"": ""https://localhost:8000/Documents/3/content"",
+      ""@odata.mediaContentType"": ""application/pdf"",
+      ""Id"": 3,
+      ""Name"": ""Sample Doc 3"",
+      ""FileLength"": 0
+    }
+  ]
+}";
+
+        private const string DOCUMENT_RESPONSE = @"{
+      ""@odata.context"": ""https://localhost:8000/$metadata#Documents/$entity"",
+      ""@odata.mediaReadLink"": ""https://localhost:8000/Documents/1/content"",
+      ""@odata.mediaContentType"": ""application/pdf"",
+      ""Id"": 1,
+      ""Name"": ""Sample Doc 1"",
+      ""FileLength"": 0
+    }";
+
+        private const string USER_RESPONSE = @"{
+    ""@odata.context"":""http://localhost:8000/$metadata#Users/$entity"",
+    ""Id"":1,
+    ""Name"":""U1""
+      }";
+
+        #endregion
+        private const string AttatchmentResponse = "Hello World!";
+
+        public DataServiceContextTrackOnlyMleTests()
+        {
+            var uri = new Uri("http://localhost:8000");
+
+            NonTrackingContext = new Container(uri)
+            {
+                MergeOption = MergeOption.NoTracking
+            };
+
+            DefaultTrackingContext = new Container(uri);
+        }
+
+        [Fact]
+        public void TestAddingBehaviourShouldNotBeModified()
+        {
+            SetupContextWithRequestPipeline(new DataServiceContext[] { DefaultTrackingContext, NonTrackingContext }, false);
+
+            var users = DefaultTrackingContext.Users;
+            var users2 = NonTrackingContext.Users;
+            Assert.Equal(users.ToList().Count, users2.ToList().Count);
+
+            // no entities should be tracked by NonTrackingContext
+            Assert.Equal(0, NonTrackingContext.EntityTracker.Entities.Count());
+            Assert.Equal(3, DefaultTrackingContext.EntityTracker.Entities.Count());
+            Assert.Equal(0, NonTrackingContext.Entities.Count);
+            Assert.Equal(3, DefaultTrackingContext.Entities.Count);
+        }
+
+        [Fact]
+        public void TrackMediaEntitiesShouldBePopulatedWithMediaEntities()
+        {
+            SetupContextWithRequestPipeline(new DataServiceContext[] { NonTrackingContext }, true);
+
+            var mleNoTracking = NonTrackingContext.Documents.ToList();
+
+            Assert.Equal(0, NonTrackingContext.EntityTracker.Entities.Count());
+
+            // verify that the stream links are equal 
+            foreach (var document in mleNoTracking)
+            {
+                var doc = document as BaseEntityType;
+                Assert.NotNull(doc.StreamDescriptor.SelfLink);
+                Assert.NotNull(NonTrackingContext.GetReadStreamUri(document));

```

Comment:
Does this actually verify that the links are equal? Or it only verifies that they are not null?

Reply:
On top of that, I think it would be beneficial to add tests that explicitly test whether the links and content type on the descriptor are the same ones that are in the response payload.